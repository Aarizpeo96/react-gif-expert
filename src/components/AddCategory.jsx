import PropTypes from "prop-types";
import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('  ')
  const setChangeValue = ({ target }) => {
    setInputValue(target.value)

  };
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return
    // onNewCategory(cat => [inputValue, ...cat]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  };
  return (
    <form onSubmit={onSubmit} aria-label="form" className="form-add-category">
      <input
        type="text"
        placeholder="Buscar keys"
        value={inputValue}
        onChange={setChangeValue} />
    </form>
  )
}

PropTypes

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}