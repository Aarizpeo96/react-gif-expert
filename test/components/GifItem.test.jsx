import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas con snapshot', () => {
    const title = 'RDR 2';
    const url = 'https://imagen/rdr2.jpg';
    test('Debe de hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})