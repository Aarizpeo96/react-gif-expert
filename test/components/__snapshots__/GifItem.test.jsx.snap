// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con snapshot Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="RDR 2"
      src="https://imagen/rdr2.jpg"
    />
    <p>
      RDR 2
    </p>
  </div>
</div>
`;
